### ISSUE TITLE
Catch all type route doesn't work as expected on a /api/{everything} reroute

### ISSUE THREAD

**Issue by marcoaam** [ThreeMammals/Ocelot#407]
## Expected Behavior

Route with /anypath/.something (note the *"."* on the path) should work and return the downstream service content.

## Actual Behavior

Using IdentityServer as an example it returns 404 and the downstream service doesn't receive the request when using the configuration specified on the steps to reproduce and requesting /identityserver/.well-known/openid-configuration.

It works and returns the expected content by adding a route like the following:

```json
    {
      "DownstreamPathTemplate": "/.well-known/openid-configuration",
      "DownstreamScheme": "http",
      "DownstreamHostAndPorts": [
        {
          "Host": "localhost",
          "Port": 5000
        }
      ],
      "UpstreamPathTemplate": "/identityserver/.well-known/openid-configuration",
      "UpstreamHttpMethod": [ "Get" ]
    }
```

Also works when adding a route with a catch all without a prefix in the path like:

```json
    {
      "DownstreamPathTemplate": "/{everything}",
      "DownstreamScheme": "http",
      "DownstreamHostAndPorts": [
        {
          "Host": "localhost",
          "Port": 5000
        }
      ],
      "UpstreamPathTemplate": "/{everything}",
      "UpstreamHttpMethod": [ "Get" ]
    }
```

## Steps to Reproduce the Problem

1 - Use IdentityServer as a downstream service and add the following config to the ocelot gateway:

```json
    {
      "DownstreamPathTemplate": "/{everything}",
      "DownstreamScheme": "http",
      "DownstreamHostAndPorts": [
        {
          "Host": "localhost",
          "Port": 5000
        }
      ],
      "UpstreamPathTemplate": "/identityserver/{everything}",
      "UpstreamHttpMethod": [ "Get" ]
    }
```

2 - Hit the /identityserver/.well-known/openid-configuration endpoint.

## Specifications

  - Version: 7.0.5
  - Platform: Windows 10 x64
  - Subsystem: .NET core 2.1

**Comment by TomPallister** [comment]
@marcoaam all good, I will have a look at this and see what's up then feedback!

**Comment by TomPallister** [comment]
released in nuget package 7.0.7 will be available on nuget soon!

**Comment by raman-m** [comment]
@AShaker01 Hi Ahmed!
Please open question discussion in [Discussions](https://github.com/ThreeMammals/Ocelot/discussions) space!
